### DISCUSSION TITLE
Family friendly edit často aktualizovaných informací na webu

### DISCUSSION THREAD

**Discussion by Grejdy** [bosancz/bosan.cz#585] [Ideas]
Napadlo mi, že když máme v interní sekci rozhraní pro výměnu oddíláků v kontaktech na webu, že by se stejným způsobem daly udělat i informace, co se mění častěji jak oddíláci - třeba hlavasové a složení oddílů na tábor (bylo by to asi užitečnější, než tam mít možnost změnit název skupiny a odkaz na tróju).
+další možný upgrade: nedalo by se to nastavit tak, aby to tahalo kontakty rovnou z databáze místo toho, aby se tam musely opisovat růčo?
Ne že bych měl v interní přístup k tomu tyhle věci měnit, ale zarazilo mne, že jsem viděl tohle rozhraní v testovací verzi (takže to jde udělat tak, aby se některé věci daly měnit bez úpravy kódu), ale že termíny a hlavasové táborů jsou v kódu té stránky napsané natvrdo, takže to očividně musí každý rok někdo otevřít a přepsat přímo ten kód.:-)

**Comment by jhavelik** [comment]
Tohle chceme pojmout nějak komplexněji a ideálně napsat celou veřejnou sekci v něčem jednodušším tak aby každý kdo má trochu vůle mohl web upravovat. Nicméně to vázne na tom že to musí někdo vzít za své vybrat vhodnou technologii a udělat to

**Reply by Grejdy** [reply]
To je možná zbytečný. Myslím, že ti, co ještě nedělaj úpravy teď, se stejně nedostanou k ničemu komplexnějšímu, než „tady máš políčko na tenhle konkrétní údaj“ a to, že to bude v něčem jednodušším (což nevím, co jednoduššího, než html máš na mysli), jim nepomůže.
V čem by to podle tebe mohlo být jednodušší? (Možná v tom tu složitost jenom nevidím, protože se v tom nehrabu dost hluboko, abych se s ní musel zabývat.)

**Reply by jhavelik** [reply]
Obecně přicházejí různé požadavky na to měnit obsah webu a pro každou věc z nich dělat administraci nedává z dlouhodobého hlediska smysl. Web je vesměs statický až na galerii a program. Do budoucna co je statické by mohlo být spravovatelne.

Nástroje na to existují a dlouhodobá vize je to do něčeho takového překlopit tak aby co je na webu statické bylo řešeno přes nějaký low code / no code nástroj.

**Reply by jhavelik** [reply]
Ještě odpověď na tvojí otázku myslím něco jednoduššího než HTML v úvahu připadá třeba WordPress nebo Jekyll (Markdown). S takovým nástrojem se naučí každý kdo má vůli něco dělat a nepotřebuje k tomu znalosti "programování"

**Reply by Grejdy** [reply]
Samozřejmě se budou objevovat požadavky na nějakou úpravu, ale jsou právě statické části webu (třeba texty o bazénu, schůzkách…), které se v průběhu času prakticky nemění, jsou věci, jako program, co se mění velmi často přes interní sekci, ale jsou i věci (právě třeba hlavasové táborů), co se sice nemění dvakrát do měsíce, ale přece jen je to dost dynamická položka, abysme to každý rok neměnili ručně za soustavného hledání, kde všude jsme zapoměli změnit rok, nebo tak něco.

**Reply by Grejdy** [reply]
Dovolili by nám tyhle systémy mít stále takhle promakanou interní sekci kompatibilní s tou veřejnou? S wordpressem jsem pracoval asi jednou, takže moc nevím, ale trošku se bojím jestli to pro nás není až příliš jednoduchý nástroj.

**Reply by jhavelik** [reply]
@Grejdy myšlenka je taková že do nějaké "user friendly" technologie by se přepsala jen veřejná část, jediné co je třeba nějak zásadně propojovat s interní sekcí je galerie a program, zbytek jsou jen statické informace, které není třeba nijak propojovat.
Na program a galerii by se musela vytvořit nějaká komponenta která by se do téhle technologie jen vložila.

**Reply by Grejdy** [reply]
To by mohlo fungovat. Přijde mi sice takový podivný mít každou sekci udělanou v něčem jiném (a trošku se obávám, že se tím potenciálně sníží počet lidí schopných upravovat interní, když nebude „na čem trénovat“, ale lidí, co by šli upravovat web bez nějaké předchozí zkušenosti, asi nebude tolik, aby to nějak extra vadilo).
Bude to teda bolest celé předělat :-).

**Reply by SmallhillCZ** [reply]
Jojo, bavili jsme se, že program a galerii bychom udělali jako webkomponenty, které by se pak vložily třeba do wordpressu no. A interní by zůstala. Hlavně aby tu veřejnou mohl upravovat kdokoliv.
